pyro/incompressible/incomp_interface.py

```diff
@@ -323,33 +275,19 @@ def riemann(ng, q_l, q_r):
         Interface state
     """
 
-    qx, qy = q_l.shape
-
-    nx = qx - 2 * ng
-    ny = qy - 2 * ng
-    ilo = ng
-    ihi = ng + nx
-    jlo = ng
-    jhi = ng + ny
-
-    s = np.zeros_like(q_l)
-
-    for i in range(ilo - 1, ihi + 1):
-        for j in range(jlo - 1, jhi + 1):
+    s = grid.scratch_array()
 
-            if (q_l[i, j] > 0.0 and q_l[i, j] + q_r[i, j] > 0.0):
-                s[i, j] = q_l[i, j]
-            elif (q_l[i, j] <= 0.0 and q_r[i, j] >= 0.0):
```

Comment:
Checking the referenced paper, it looks like this condition should be `q_l <= 0 or q_r >= 0`, not and

Reply:
Maybe not. Switching to `or` gives some weird artifacts: ![incompressible_or](https://user-images.githubusercontent.com/522590/227578824-4a0f62a7-42ae-4f1d-9375-087bdc400bbe.png)